gc.c

```diff
@@ -8944,16 +8906,11 @@ gc_writebarrier_incremental(VALUE a, VALUE b, rb_objspace_t *objspace)
         }
         else if (RVALUE_OLD_P(a) && !RVALUE_OLD_P(b)) {
             if (!RVALUE_WB_UNPROTECTED(b)) {
-                gc_report(1, objspace, "gc_writebarrier_incremental: [GN] %p -> %s\n", (void *)a, obj_info(b));
-                RVALUE_AGE_SET_OLD(objspace, b);
-
-                if (RVALUE_BLACK_P(b)) {
-                    gc_grey(objspace, b);
-                }
+                rgengc_remember(objspace, a);
             }
             else {
```

Comment:
Similarly here the only difference between the two branches is one calls `gc_report`. I think it could be simplified.